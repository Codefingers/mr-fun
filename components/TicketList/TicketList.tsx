import React, { useState, Component } from 'react';
import styled from 'styled-components';

import { Product } from '../../types';

const TListC = styled.div`
  margin: 16px 24px;
  padding: 16px 24px;
`;

const TicketCardContainer = styled.div`
  padding: 16px 24px 24px;
  border: 4px solid #1D2433;
  margin-bottom: 16px;
  display: flex;
  flex-direction: column;

  h3 {
    margin: 0;
  }

  p {
    padding-bottom: 16px;
  }

  &.active {
    border-color: ${({ theme }) => theme.colors.accent};
  }

  button {
    align-self: flex-end;
  }
`;

export const TicketCard: React.FC<{ product: Product }> = (props) => {
  const [active, setActive] = useState<boolean>();
  const {
    product: { id, title, price },
  } = props;
  const activeClass = active ? 'card active' : 'card';

  return (
    <TicketCardContainer
      className={activeClass}
      onClick={() => {
        setActive(true);
      }}
    >
      <h3>{title}</h3>
      <p>Â£{price}</p>
      <button>Add to Basket</button>
    </TicketCardContainer>
  );
};


export class TicketList extends Component<{
  products: Array<Product>;
  title: string;
}> {
  render() {
    const { products, title } = this.props;

    return (
      <TListC>
        <h1>What passes will you need?</h1>
        <h2>{title}</h2>
        <div>
          {products.map((product, index) => (
            <div key={`listing-${index}`}>
              <TicketCard product={product} />
            </div>
          ))}
        </div>
      </TListC>
    );
  }
}

export default TicketList;
