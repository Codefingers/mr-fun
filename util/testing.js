import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={{
        colors: {
          body: '#838fa6',
          heading: '#ebedf2',
          accent: '#BA99FF',
          button: '#FFF',
        },
      }}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
