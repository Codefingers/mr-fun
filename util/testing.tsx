import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={theme}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
