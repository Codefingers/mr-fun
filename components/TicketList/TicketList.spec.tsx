import React from 'react';
import { screen, render, fireEvent } from '../../util/testing';

import { TicketList, TicketCard } from './TicketList';

describe('<TicketList />', () => {
  const props = {
    title: 'test title',
    products: [
      {
        id: '1',
        title: 'prod 1',
        price: 10.0,
        quantity: 2,
      },
      {
        id: '2',
        title: 'prod 2',
        price: 12.4,
        quantity: 1,
      },
    ],
  };

  it('renders', () => {
    const { container } = render(<TicketList {...props} />);
    expect(container).toMatchSnapshot();
  });
});

describe('<TicketCard />', () => {
  const props = {
    product: {
      id: '1',
      title: 'Product 1',
      price: 10.0,
      quantity: 2,
    },
  };

  it('renders', () => {
    const { container } = render(<TicketCard {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('calls addToBasket with product info', () => {
    render(<TicketCard {...props} />);
    fireEvent.click(screen.getByText(/Product 1/i));
  });
});
