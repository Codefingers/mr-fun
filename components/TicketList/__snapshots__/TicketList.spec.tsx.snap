// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketCard /> renders 1`] = `
<div>
  <div
    class="card TicketList__TicketCardContainer-sc-sf3b3b-1 fCyvOu"
  >
    <h3>
      Product 1
    </h3>
    <p>
      £
      10
    </p>
    <button>
      Add to Basket
    </button>
  </div>
</div>
`;

exports[`<TicketList /> renders 1`] = `
<div>
  <div
    class="TicketList__TListC-sc-sf3b3b-0 ciLCpI"
  >
    <h1>
      What passes will you need?
    </h1>
    <h2>
      test title
    </h2>
    <div>
      <div>
        <div
          class="card TicketList__TicketCardContainer-sc-sf3b3b-1 fCyvOu"
        >
          <h3>
            prod 1
          </h3>
          <p>
            £
            10
          </p>
          <button>
            Add to Basket
          </button>
        </div>
      </div>
      <div>
        <div
          class="card TicketList__TicketCardContainer-sc-sf3b3b-1 fCyvOu"
        >
          <h3>
            prod 2
          </h3>
          <p>
            £
            12.4
          </p>
          <button>
            Add to Basket
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
